SupermarketMap.py

```diff
@@ -0,0 +1,133 @@
+        """returns the array for a given tile character"""
+        if char == "#":
+            return self.extract_tile(0, 0)
+        elif char == "Z":
+            return self.extract_tile(0, 4)
+        elif char == "C":
+            return self.extract_tile(8, 3)
+        elif char == "D":
+            return self.extract_tile(3, 13)
+        elif char == "E":
+            return self.extract_tile(8, 7)
+        elif char == "S":
+            return self.extract_tile(5, 8)     
+        elif char == "V":
+            return self.extract_tile(2, 11)        
+        elif char == "X":
+            return self.extract_tile(8, 14)
+        else:
+            return self.extract_tile(1, 2)
+
+           
+        
+    def prepare_map(self):
+        """prepares the entire image as a big numpy array"""
+        for row, line in enumerate(self.contents):
+            for col, char in enumerate(line):
+                bm = self.get_tile(char)
+                y = row*TILE_SIZE
+                x = col*TILE_SIZE
+                self.image[y:y+TILE_SIZE, x:x+TILE_SIZE] = bm
+
+    def draw(self, frame):
+        """
+        draws the image into a frame
+        """
+        frame[0:self.image.shape[0], 0:self.image.shape[1]] = self.image
+
+    def write_image(self, filename):
+        """writes the image into a file"""
+        cv2.imwrite(filename, self.image)
+    """Constructor for the map, visualizes the supermarket layout(MARKET) with extracting the icons from the tiles"""
+
+    def __init__(self, layout, tiles):
+        """
+        layout : a string with each character representing a tile
+        tiles   : a numpy array containing all the tile images
+        """
+        self.tiles = tiles
+        # split the layout string into a two dimensional matrix
+        self.contents = [list(row) for row in layout.split("\n")]
+        self.ncols = len(self.contents[0])
+        self.nrows = len(self.contents)
+        self.image = np.zeros(
+            (self.nrows*TILE_SIZE, self.ncols*TILE_SIZE, 3), dtype=np.uint8
+        )
+        self.prepare_map()
+
+    def extract_tile(self, row, col):
+        """extract a tile array from the tiles image"""
+        y = row*TILE_SIZE
+        x = col*TILE_SIZE
+        return self.tiles[y:y+TILE_SIZE, x:x+TILE_SIZE]
+
+    def get_tile(self, char):
+        """returns the array for a given tile character"""
+        if char == "#":
+            return self.extract_tile(0, 0)
+        elif char == "Z":
+            return self.extract_tile(0, 4)
+        elif char == "C":
+            return self.extract_tile(5, 3)
+        elif char == "D":
+            return self.extract_tile(3, 13)
+        elif char == "E":
+            return self.extract_tile(7, 3)
+        elif char == "S":
+            return self.extract_tile(5, 8)     
+        elif char == "V":
+            return self.extract_tile(2, 11)        
+        elif char == "X":
+            return self.extract_tile(6, 10)
+        else:
```

Comment:
The icon of the titles assigned to E, X and C have to checked. 
Coordinates for E (8,3)
Coordinates for X (8,14)
Coordinates for C (8,3)